server/src/test/java/org/elasticsearch/index/mapper/TextFieldMapperTests.java

```diff
@@ -1227,4 +1227,27 @@ public void testIgnoreEagerGlobalOrdinalsOnLegacyIndex() throws IOException {
         mapperService = createMapperService(Version.fromString("5.0.0"), mapping);
         assertFalse(((TextFieldMapper) mapperService.documentMapper().mappers().getMapper("field")).fieldType().eagerGlobalOrdinals());
     }
+
+    public void testDocValues() throws IOException {
+        MapperService mapper = createMapperService(fieldMapping(b -> b.field("type", "text")));
+        assertScriptDocValues(mapper, "foo", equalTo(List.of("foo")));
```

Comment:
I think it's worth testing that an input that would otherwise be tokenized is returned whole here. Basically test that we get "foo bar" back, rather than a list of "foo" and "bar".